bokeh/themes/theme.py

```diff
@@ -0,0 +1,83 @@
+from __future__ import absolute_import, print_function
+
+import yaml
+
+from ..properties import HasProps
+
+# whenever we cache that there's nothing themed for a class, we
+# use this same dict instance, so we don't have a zillion empty
+# dicts in our caches.
+_empty_dict = dict()
+
+class Theme(object):
```

Comment:
one thing I was just thinking about, currently for DirectoryHandler I share the Theme object among all created sessions without copying it, which assumes Theme is immutable. If we add any "setters" to Theme, then it would need to be copied into each session. Which is probably fine but I may add a comment for future reference on this.